### PR TITLE
Remove CronJobControllerV2 feature gate

### PR DESCRIPTION
**Pull request by soltysh**
Part of https://github.com/kubernetes/enhancements/issues/19

/assign @chrisnegus @pacoxu @alaypatel07

### CODE REVIEW

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -557,10 +554,10 @@ Each feature gate is designed for enabling/disabling a specific feature:
...
    for more details.
-- `ControllerManagerLeaderMigration`: Enables Leader Migration for 
-  [kube-controller-manager](/docs/tasks/administer-cluster/controller-manager-leader-migration/#initial-leader-migration-configuration) and 
-  [cloud-controller-manager](/docs/tasks/administer-cluster/controller-manager-leader-migration/#deploy-cloud-controller-manager) which allows a cluster operator to live migrate 
-  controllers from the kube-controller-manager into an external controller-manager 
+- `ControllerManagerLeaderMigration`: Enables Leader Migration for
+  [kube-controller-manager](/docs/tasks/administer-cluster/controller-manager-leader-migration/#initial-leader-migration-configuration) and
+  [cloud-controller-manager](/docs/tasks/administer-cluster/controller-manager-leader-migration/#deploy-cloud-controller-manager) which allows a cluster operator to live migrate
+  controllers from the kube-controller-manager into an external controller-manager
   (e.g. the cloud-controller-manager) in an HA cluster without downtime.
```
Comment (pacoxu) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:561]: L564 needs to be removed as well.

**Review by pacoxu — COMMENTED** [review]
/lgtm

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 2]
```diff
@@ -255,9 +255,6 @@ different Kubernetes components.
 | `CSIServiceAccountToken` | `false` | Alpha | 1.20 | 1.20 |
 | `CSIServiceAccountToken` | `true` | Beta | 1.21 | 1.21 |
 | `CSIServiceAccountToken` | `true` | GA | 1.22 | |
-| `CronJobControllerV2` | `false` | Alpha | 1.20 | 1.20 |
-| `CronJobControllerV2` | `true` | Beta | 1.21 | 1.21 |
-| `CronJobControllerV2` | `true` | GA | 1.22 | - |
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:260]: I don't think we should remove this.

Comment (soltysh) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:260]: @tengqm I did wonder about that too, but on the other hand that value is not present anymore, so you can't set it. It'll make senese for previous versions of k8s, but those have their own docs, that's why I've decided to go with it.

**Comment by soltysh** [comment]
Rebased.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md (hunk 2 above)
Comment (reylejano) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:261]: I agree with @tengqm , these are left in even though we maintain docs for the current + past 4 releases

Comment (soltysh) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:261]: Fair, I'm closing the PR in that case.

**Comment by chrisnegus** [comment]
@reylejano I'm going to look for a place for us to document that people reviewing release docs not remove content from the feature gate from previous releases, even if the feature no longer exists in the current release. Any suggestions where might be a good place to document that?

**Comment by reylejano** [comment]
> @reylejano I'm going to look for a place for us to document that people reviewing release docs not remove content from the feature gate from previous releases, even if the feature no longer exists in the current release. Any suggestions where might be a good place to document that?

It's already stated in:
https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review

> If your feature is an Alpha or Beta feature and is behind a feature gate, make sure you add it to Alpha/Beta Feature gates table as part of your pull request. With new feature gates, a description of the feature gate is also required. If your feature is GA'ed or deprecated, make sure to move it from that table to Feature gates for graduated or deprecated features table with Alpha and Beta history intact.

**Pull request closed without merging**